### DISCUSSION TITLE
MacCatalyst sample app error: Could not find the task 'Uno: net8.0-maccatalyst | Debug | maccatalyst-arm64'

### DISCUSSION THREAD

**Discussion by leonschoi** [unoplatform/uno#14932] [Q&A]
I am trying to run a sample app on a Mac with MacCatalyst, created with the command

```
dotnet new unoapp -o MyApp --preset=blank -platforms android maccatalyst wasm --vscode
```
Android and wasm runs, but with MacCatalyst it gives the error
```
Could not find the task 'Uno: net8.0-maccatalyst | Debug | maccatalyst-arm64'
```

In `.vscode/tasks.json` the `"label"` specifies `"Uno: net8.0-maccatalyst | Debug | maccatalyst-arm64"`
```
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "Uno",
      "targetFramework": "net8.0-maccatalyst",
      "args": [
        "\"${workspaceFolder}/MyApp.Mobile/MyApp.Mobile.csproj\"",
        "-f:net8.0-maccatalyst",
        "-c:Debug",
        "-r:maccatalyst-arm64",
        "-clp:NoSummary",
        "-p:GenerateFullPaths=true",
        "-p:UnoForceSingleTFM=true",
        "-bl:\"${workspaceFolder}/MyApp.Mobile/net8.0-maccatalyst-Debug-maccatalyst-arm64.binlog\""
      ],
      "problemMatcher": [
        "$msCompile"
      ],
      "label": "Uno: net8.0-maccatalyst | Debug | maccatalyst-arm64"
    }
    ...
  ]
}
```
So the task is specified but cannot be found. How can I make it recognize the task specified?

**Comment by spouliot** [comment]
That should work but, without knowing more about your setup and versions, it's hard to guess what could be wrong.

1. Check your uno environment with `uno-check`
https://platform.uno/docs/articles/get-started-vscode.html?tabs=windows%2Cubuntu1804#check-your-environment
That will also (ask you to) install the latest version of the templates.

2. Paste the logs here as it will tell us more about what you're using.

3. Make sure to update VS Code and use the latest version (0.12 now) of the Uno Platform extension.

4. You should not need to add a specific task for Catalyst - the extension will create one on the fly with your TFM, configuration and device (so it match the what you have selected in the UI).
The template also generates a _generic_ Mobile launch configuration. You must select this `Uno Platform Mobile` in the debugger before pressing `F5`.
